virttest/utils_net.py

```diff
@@ -4468,3 +4468,88 @@ def delete_linux_bridge_tmux(linux_bridge_name, iface_name=None, ignore_status=F
     else:
         cmd = 'ip link delete %s' % linux_bridge_name
     return utils_misc.cmd_status_output(cmd, shell=True, verbose=True, ignore_status=ignore_status)
+
+
+def check_filter_rules(ifname, bandwidth, expect_none=False):
+    """
+    Check bandwidth settings via 'tc filter' output
+
+    :param ifname: the interface name
+    :param bandwidth: the settings in the xml
+    :param expect_none: whether or not expect nothing in output,
+                        default to be False
+    :return: return True if get expected result, else return False
+    """
+    cmd = "tc -d filter show dev %s parent ffff:" % ifname
+    filter_output = process.run(cmd, shell=True).stdout_text
+    logging.debug("Bandwidth filter output: %s", filter_output)
+    if expect_none:
+        return not filter_output.strip()
+    if not filter_output.count("filter protocol all pref"):
+        logging.error("Can't find 'protocol all' settings in filter rules")
+        return False
+    filter_pattern = r".*police.*rate (\d+)(K?M?)bit burst (\d+)(K?M?)b.*"
+    se = re.search(r"%s" % filter_pattern, filter_output, re.M)
```

Comment:
I think it'll be great if you can update 'se' to a more meaningful name.

Reply:
Updated to be tc_police.